shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -285,13 +290,35 @@ public static String getScopedKeyValueTableName(String scope, String streamName)
         return getScopedStreamNameInternal(scope, streamName).toString();
     }
 
+    /**
+     * Method to generate Fully Qualified  TableSegmentName using scope, stream and segment id.
+     * Table segments for KVTables have segment names suffixed with _table,
+     * so we can have a stream and KeyValueTable with the same name.
+     *
+     * @param scope scope to be used in the ScopedTableSegment name
+     * @param kvTableName kvTable name to be used in ScopedTableSegment name.
+     * @param segmentId segment id to be used in ScopedStreamSegment name.
+     * @return fully qualified TableSegmentName for a TableSegment that is part of the KeyValueTable.
+     */
+    public static String getQualifiedTableSegmentName(String scope, String kvTableName, long segmentId) {
+        int segmentNumber = getSegmentNumber(segmentId);
+        int epoch = getEpoch(segmentId);
+        StringBuffer sb = getScopedStreamNameInternal(scope, kvTableName);
+        sb.append('/');
+        sb.append(segmentNumber);
+        sb.append(EPOCH_DELIMITER);
+        sb.append(epoch);
+        sb.append(KVTABLE_SUFFIX);
```

Comment:
Why do we need to do this? Virtually all Segments for a KV Table will have this suffix and they'll all be in the same directory. So this is useless in my view.

Reply:
This suffix is to distinguish KVTable Segments from Stream Segments. A KVTable and Stream can have the same name and be inside the same scope and this suffix makes sure we don;t mix up the segment names for the two.